faust/stores/bigtable.py

```diff
@@ -0,0 +1,605 @@
+        return partitions
+
+    def _get_current_partitions(self) -> List[int]:
+        event = current_event()
+        if (
+            event is not None
+            and event.message.topic is not None
+            and not self.table.is_global
+            and not self.table.use_partitioner
+        ):
+            partition = event.message.partition
+            return [partition]
+        return self._active_partitions()
+
+    def _get_partitions_for_key(self, key: bytes) -> List[int]:
+        if key in self._key_index:
+            return [self._key_index[key]]
+        return self._get_current_partitions()
+
+    @staticmethod
+    def bigtable_exrtact_row_data(row_data):
+        return list(row_data.to_dict().values())[0][0].value
+
+    def _del_cache(self, key: bytes):
+        if self._startup_cache is not None:
+            self._startup_cache[key] = None
+
+    def _set_cache(self, key: bytes, value):
+        if self._startup_cache is not None:
+            self._startup_cache[key] = value
+
+    def _get_cache(self, key: bytes):
+        if self._startup_cache_enable and self._startup_cache is not None:
+            if key in self._startup_cache:
+                return self._startup_cache[key], True
+        return None, False
+
+    def _invalidate_startup_cache(self):
+        if self._startup_cache is not None:
+            self._startup_cache.clear()
+            self._startup_cache_partitions = set()
+            gc.collect()
+            self.log.info(f"Invalidated startup cache for table {self.table_name}")
+        self._invalidation_timer.cancel()
+        del self._invalidation_timer
+        self._invalidation_timer = None
+
+    def _set_mutation(self, mutated_row: DirectRow):
+        self._mutation_batcher.mutate(mutated_row)
+
+    def _bigtable_get(self, keys: List[bytes]) -> Tuple[Optional[bytes], Optional[int]]:
+        rowset = BT.RowSet()
+        for key in keys:
+            rowset.add_row_key(key)
+        if self._mutation_batcher_enable:
+            self._mutation_batcher.flush()
+
+        rows = self.bt_table.read_rows(row_set=rowset, filter_=self.row_filter)
+        for row in rows:
+            if row is not None:
+                partition = self._get_partition_from_bigtable_key(row.row_key)
+                return self.bigtable_exrtact_row_data(row), partition
+        return None, None
+
+    def _get(self, key: bytes) -> Optional[bytes]:
+        try:
+            value, found = self._get_cache(key)
+            if found:
+                return value
+
+            partitions = set(self._get_partitions_for_key(key))
+            # Remove partitions that we already have in cache
+            partitions.difference_update(self._startup_cache_partitions)
+            # Nothing todo
+            if len(partitions) == 0:
+                return None
+
+            keys = [self._add_partition_prefix_to_key(key, p) for p in partitions]
+            if len(keys) == 0:
```

Comment:
this check is not needed since we already check the length of the partitions list four lines above.